### PR TITLE
feat(bigquery): Add example for creating a reservation assignment

### PR DESCRIPTION
**Pull request by ccarpentiere**
## Description

Fixes #269302678

Note: If you are not associated with Google, open an issue for discussion before submitting a pull request.

## Checklist

**Readiness**

- [X] Yes, **merge** this PR after it is approved
- [ ] No, don't **merge** this PR after it is approved

**Style**

- [X] My sample follows the rules described for Terraform in the [Effective Samples style guide](https://googlecloudplatform.github.io/samples-style-guide/)
- [X] My sample follows the other requirements and best practices in the [Contributing
guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#other-requirements-and-best-practices)

**Testing**

- [X] I have performed tests described in the [Contributing guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md):

   - [X] **[Tests](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#set-up-the-test-environment)** pass: `terraform apply`
   - [X] **[Lint](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#linting-and-formatting)** pass: `terraform fmt` check

**Intended location**

- [X] Yes, this sample will be (or already is) included on cloud.google.com
      Location(s): https://cloud.google.com/bigquery/docs/reservations-get-started#assign_a_project_to_a_reservation

- [ ] No, this sample won't be included on cloud.google.com
      Reason:

**API enablement**

- [ ] If the sample needs an API enabled to pass testing, I have added the service to the [Test setup file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/test/setup/main.tf)

**Review**

- [ ] If this sample adds a new directory, I have added codeowners to the [CODEOWNERS file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/.github/CODEOWNERS)

### CODE REVIEW

**Review by wj-chen — APPROVED** [review]
I'm not familiar with how test.yaml in this report but main.tf LGTM.

**Comment by ccarpentiere** [comment]
Can I merge this since I have one approval, or do I need to wait for review from terraform-google-modules/terraform-samples-reviewers?

File: bigquery/bigquery_create_reservation/test.yaml
Comment (wj-chen) [inline bigquery/bigquery_create_reservation/test.yaml:1]: 2024?

**Comment by wj-chen** [comment]
> Can I merge this since I have one approval, or do I need to wait for review from terraform-google-modules/terraform-samples-reviewers?

Would suggest to get an approval from the other group since bigquery-terraform-swe is primarily to check the validity of the content of the examples and not so much on the style guide and other repo-specific guidelines.

File: bigquery/bigquery_create_reservation/test.yaml
Comment (glasnt) [inline bigquery/bigquery_create_reservation/test.yaml:1]: Adding this file will no longer test the bq reservation resource. Is this intended?

File: bigquery/bigquery_create_reservation/main.tf
Comment (glasnt) [inline bigquery/bigquery_create_reservation/main.tf:33]: This value should be based on a variable, not hardcoded to a value. This config should work: 

```suggestion
data "google_project" "project" {}

resource "google_bigquery_reservation_assignment" "default" {
  assignee    = "projects/${data.google_project.project.project_id}"
```

File: bigquery/bigquery_create_reservation/test.yaml
Comment (ccarpentiere) [reply bigquery/bigquery_create_reservation/test.yaml:1]: This sample was originally published in 2023, so I left the copyright info as-is. Do we have to update it when we make sample updates?

File: bigquery/bigquery_create_reservation/main.tf
Comment (ccarpentiere) [reply bigquery/bigquery_create_reservation/main.tf:33]: I'm fine this change if using variables is now the guidance.

File: bigquery/bigquery_create_reservation/test.yaml
Comment (wj-chen) [reply bigquery/bigquery_create_reservation/test.yaml:1]: Looks like  a new file to me so I'd timestamp it by creation date

**Review by glasnt — APPROVED** [review]

**Pull request merged**
